### PR TITLE
test: add tests for reservation utils

### PR DESCRIPTION
**Pull request by saintube**
Signed-off-by: saintube <saintube@foxmail.com>

### Ⅰ. Describe what this PR does

add unit tests for reservation utility functions

### V. Checklist

- [X] I have written necessary docs and comments
- [X] I have added necessary unit tests and integration tests
- [X] All checks passed in `make test`

### CODE REVIEW

**Comment by jasonliu747** [comment]
Can you also add some UTs for `IsReservationAvailable`, `IsReservationSucceeded`, `IsReservationFailed`, and `IsReservationExpired`? Thanks.

File: pkg/util/reservation_test.go [hunk 1]
```diff
@@ -258,3 +259,318 @@ func TestIsObjValidActiveReservation(t *testing.T) {
...
+			PodHandler: &fakePodHandler{
+				t: t,
+			},
+		}
+
+		h.OnAdd(testReservation)
+
+		h.OnUpdate(testReservation, testReservation)
+
+		h.OnDelete(testReservation)
```
Comment (jasonliu747) [inline pkg/util/reservation_test.go:567]: could also add a case for `cache.DeletedFinalStateUnknown`

Comment (saintube) [reply pkg/util/reservation_test.go:709]: done

**Review by jasonliu747 — APPROVED** [review]
/lgtm

**Pull request merged**